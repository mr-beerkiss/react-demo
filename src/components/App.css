:local(.App) {
  text-align: center;
}

:local(.App__header) {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

:local(.App__header__logo) {
  height: 80px;
}
