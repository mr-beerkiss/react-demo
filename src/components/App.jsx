import React, { Component } from 'react'
import logo from '../assets/logo.svg'
import styles from './App.css'
import TaskList from './taskList/TaskList.jsx'

class App extends Component {
  constructor () {
    super()

    this.state = {
      tasks: [{
        name: 'Task 1',
        complete: false,
        archived: false
      }]
    }
  }

  render () {
    return (
      <div className={styles.app}>
        <div className={styles.header}>
          <img
            src={logo}
            className={styles.logo}
            alt='logo' />
          <h2>Welcome to Damon's Demo Task List</h2>
        </div>
        <TaskList
          data={this.state.tasks}
          onTaskChange={this.taskChangeHandler}
          onTaskComplete={this.taskClickHandler}
          onTaskArchive={this.taskArchvieHandler}
          onNewTask={this.newTaskHandler} />
      </div>
    )
  }

  taskChangeHandler = (index, value) => {
    this.updateTask('name', index, value)
  }

  taskClickHandler = (index, complete) => {
    this.updateTask('complete', index, complete)
  }

  taskArchvieHandler = (index, archive) => {
    this.updateTask('archived', index, archive)
  }

  updateTask (field, index, value) {
    const newTasks = this.state.tasks.concat([]) // immutability
    newTasks[index][field] = value
    this.setState({
      tasks: newTasks
    })
  }

  newTaskHandler = () => {
    const newTasks = this.state.tasks.concat([{
      name: '',
      complete: false,
      archived: false
    }]) // immutability
    this.setState({
      tasks: newTasks
    })
  }
}

export default App
