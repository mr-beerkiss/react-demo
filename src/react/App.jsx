import React, { Component } from 'react'
import logo from '../assets/logo.svg'
import styles from './App.css'
import TaskList from './components/taskList/TaskList.jsx'
import connectState from './App.connect'

export class App extends Component {
  render () {
    const {tasks} = this.props

    return (
      <div className={styles.app}>
        <div className={styles.header}>
          <img
            src={logo}
            className={styles.logo}
            alt='logo' />
          <h2>Welcome to Damon's Demo Task List</h2>
        </div>
        <TaskList
          data={tasks}
          onTaskChange={this.taskChangeHandler}
          onTaskComplete={this.taskCompleteHandler}
          onTaskRemove={this.taskRemoveHandler}
          onAddTask={this.addTaskHandler} />
      </div>
    )
  }

  addTaskHandler = () => {
    this.props.addTaskAction()
  }

  taskChangeHandler = (index, value) => {
    this.props.taskChangeAction(index, value)
  }

  taskCompleteHandler = (index, complete) => {
    this.props.taskCompleteAction(index, complete)
  }

  taskRemoveHandler = (index) => {
    this.props.taskRemoveAction(index)
  }
}

export default connectState(App)
