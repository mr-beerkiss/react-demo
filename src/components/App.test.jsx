/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'
import App from './App.jsx'

describe('<App />', () => {
  it('snapshot: render header and title', () => {
    const tree = renderer.create(
      <App />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
