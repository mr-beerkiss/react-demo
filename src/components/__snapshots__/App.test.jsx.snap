exports[`<App /> snapshot: render header and title 1`] = `
<div
  className="App">
  <div
    className="App__header">
    <img
      alt="logo"
      className="App__header__logo"
      src="logo.svg" />
    <h2>
      Welcome to Damon\'s Demo Task List
    </h2>
  </div>
</div>
`;
